tools/harness-simulation/posix/otbr_sim/otbr_docker.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python3
+#
+#  Copyright (c) 2022, The OpenThread Authors.
+#  All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are met:
+#  1. Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#  2. Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#  3. Neither the name of the copyright holder nor the
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+import logging
+import os
+import subprocess
+import time
+
+from config import (OT_PATH, OTBR_DOCKER_IMAGE)
+
+
+class OtbrDocker:
+
+    def __init__(self, nodeid: int, docker_name: str):
+        self.nodeid = nodeid
+        self.docker_name = docker_name
+
+        self.logger = logging.getLogger('otbr_docker.OtbrDocker')
+        self.logger.setLevel(logging.INFO)
+
+        self._socat_proc = None
+        self._ot_rcp_proc = None
+
+        self._rcp_device_pty = None
+        self._rcp_device = None
+
+        self._launch()
+
+    def __repr__(self) -> str:
+        return f'OTBR<{self.nodeid}>'
+
+    def _launch(self):
+        self.logger.info('Launching %r ...', self)
```

Comment:
Since this file is interpreted by python3, I'm slightly inclined to use python3 formatted strings.

Reply:
`self.logger` is not a self-written class. It is generated by `logging.getLogger()`. The arguments of the method `self.logger.info` is `msg: object, *args: object, ...`.

I think it is better to keep the intended usage when using `logging` library.